File: src/main/java/ru/mdashlw/hypixel/ranked/reply/impl/SeasonsReply.java
package ru.mdashlw.hypixel.ranked.reply.impl;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.mdashlw.hypixel.ranked.entities.RankedSeason;
import ru.mdashlw.hypixel.ranked.reply.Reply;

import java.util.List;

public class SeasonsReply extends Reply<List<RankedSeason>> {
    @JsonProperty
    private List<RankedSeason> seasons;

    @Override
    public List<RankedSeason> getValue() {
        return seasons;
    }
}

File: src/main/kotlin/ru/mdashlw/hypixel/api/ranked/reply/impl/SeasonsReply.kt
package ru.mdashlw.hypixel.api.ranked.reply.impl

import ru.mdashlw.hypixel.api.ranked.entities.Season
import ru.mdashlw.hypixel.api.ranked.reply.Reply

class SeasonsReply(
    success: Boolean,
    cause: String?,
    seasons: List<Season>?
) : Reply<List<Season>>(success, cause, seasons)
